openmc/filter.py

```diff
@@ -532,19 +554,48 @@ class CellFilter(IntegralFilter):
         filter's bins.
 
     """
+    @property
+    def bins(self):
+        return self._bins
 
+    @property
+    def num_bins(self):
+        return len(self.bins)
 
-class CellbornFilter(IntegralFilter):
-    """Bins tally events based on the cell that the particle was born in.
+    @bins.setter
+    def bins(self, bins):
+        # Format the bins as a 1D numpy array.
+        bins = np.atleast_1d(bins)
+
+        # Check the bin values.
+        cv.check_iterable_type('filter bins', bins, (Integral, openmc.Cell))
+        for edge in bins:
+            if isinstance(edge, Integral):
+                cv.check_greater_than('filter bin', edge, 0, equality=True)
+
+        # Extract id values.
+        if any([isinstance(b, openmc.Cell) for b in bins]):
```

Comment:
What do you think about simplifying matters and putting lines 571-574 in a setter for the base `Filter` class? Then each subclass would simply include lines 568 and 577-581.